File: data/code_pairs/fortran/2377.f90
module performance
  implicit none
contains
  subroutine perf_header()
    write(*,*)
    write(*,'("        # Photons    CPU time (sec)    Photons/sec  ")')
    write(*,'("      ----------------------------------------------")')
  end subroutine perf_header

  subroutine perf_footer()
    write(*,'("      ----------------------------------------------")')
    write(*,*)
  end subroutine perf_footer

  subroutine perf_numbers(count, time)
    integer, intent(in) :: count
    real, intent(in) :: time
    if(time <= 0.0) then
      write(*,'(1X,A,I12,3X,F10.1,2X,A)') " ", count, " ", time, "   ...   "
    else
      write(*,'(1X,A,I12,3X,F10.1,2X,F9.2)') " ", count, " ", time, real(count)/time
    end if
  end subroutine perf_numbers
end module performance

! Test Program
program test_performance
  use performance
  call perf_header()
  call perf_numbers(1000000, 123.4)
  call perf_numbers(500000, 0.0)
  call perf_footer()
end program test_performance

File: data/code_pairs/cpp/2377.cpp
#include <iostream>
#include <iomanip>
#include <limits>

namespace performance {
    void perf_header() {
        std::cout << std::endl;
        std::cout << "        # Photons    CPU time (sec)    Photons/sec  " << std::endl;
        std::cout << "      ----------------------------------------------" << std::endl;
    }

    void perf_footer() {
        std::cout << "      ----------------------------------------------" << std::endl;
        std::cout << std::endl;
    }

    void perf_numbers(int count, double time) {
        std::cout << " ";
        if (time <= 0.0) {
            std::cout << std::setw(12) << count 
                      << std::setw(8) << std::fixed << std::setprecision(1) << time 
                      << "       ...   " << std::endl;
        } else {
            std::cout << std::setw(12) << count 
                      << std::setw(8) << std::fixed << std::setprecision(1) << time 
                      << std::setw(12) << std::fixed << std::setprecision(2) << count / time << std::endl;
        }
    }
}

int main() {
    performance::perf_header();
    performance::perf_numbers(1000000, 123.4);
    performance::perf_numbers(500000, 0.0);
    performance::perf_footer();
    return 0;
}